src/main/java/org/testng/internal/TestMethodWithDataProviderMethodWorker.java

```diff
@@ -127,13 +127,7 @@ public TestMethodWithDataProviderMethodWorker(
       if (m_failureCount > 0 && m_skipFailedInvocationCounts) {
         while (m_invocationCount-- > 0) {
           ITestResult r =
-              new TestResult(
-                  m_testMethod.getTestClass(),
-                  m_testMethod,
-                  null,
-                  start,
-                  System.currentTimeMillis(),
-                  m_testContext);
+              TestResult.newEndTimeAwareTestResult(m_testMethod, null, start, m_testContext);
```

Comment:
Instead of `null`, maybe we can have a custom Exception like TooMuchInvoicationCountException

Reply:
What would we be achieving by introducing a new exception ?